pulp/apis/cuopt_api.py

```diff
@@ -0,0 +1,337 @@
+import ctypes
+import os
+import subprocess
+import sys
+import warnings
+from uuid import uuid4
+import numpy as np
+from ..constants import (
+    LpBinary,
+    LpConstraintEQ,
+    LpConstraintGE,
+    LpConstraintLE,
+    LpContinuous,
+    LpInteger,
+    LpMaximize,
+    LpMinimize,
+    LpStatusInfeasible,
+    LpStatusNotSolved,
+    LpStatusOptimal,
+    LpStatusUnbounded,
+    LpStatusUndefined,
+)
+from .core import (
+    LpSolver,
+    LpSolver_CMD,
+    PulpSolverError,
+    clock,
+    ctypesArrayFill,
+    sparse,
+)
+
+# COPT string convention
+if sys.version_info >= (3, 0):
+    coptstr = lambda x: bytes(x, "utf-8")
+else:
+    coptstr = lambda x: x
```

Comment:
PuLP requires python >= 3.9 so no need to add this conditional

Reply:
Removed in #839